langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiResponsesStreamingChatModel.java

```diff
@@ -0,0 +1,972 @@
+        } else {
+            throw new IllegalArgumentException("Image must have either url or base64Data");
+        }
+    }
+
+    private String toToolChoiceString(dev.langchain4j.model.chat.request.ToolChoice toolChoice) {
+        if (toolChoice == null) {
+            return null;
+        }
+        return switch (toolChoice) {
+            case AUTO -> "auto";
+            case REQUIRED -> "required";
+            case NONE -> "none";
+        };
+    }
+
+    private Map<String, Object> toResponseTextConfig(dev.langchain4j.model.chat.request.ResponseFormat responseFormat) {
+        if (responseFormat == null
+                || responseFormat.type() == dev.langchain4j.model.chat.request.ResponseFormatType.TEXT) {
+            return null;
+        }
+
+        var textConfig = new HashMap<String, Object>();
+        dev.langchain4j.model.chat.request.json.JsonSchema jsonSchema = responseFormat.jsonSchema();
+
+        if (jsonSchema == null) {
+            var format = new HashMap<String, Object>();
+            format.put(FIELD_TYPE, TYPE_JSON_OBJECT);
+            textConfig.put(FIELD_FORMAT, format);
+        } else {
+            if (!(jsonSchema.rootElement() instanceof dev.langchain4j.model.chat.request.json.JsonObjectSchema
+                    || jsonSchema.rootElement() instanceof dev.langchain4j.model.chat.request.json.JsonRawSchema)) {
+                throw new IllegalArgumentException(
+                        "For OpenAI Responses API, the root element of the JSON Schema must be either a JsonObjectSchema or a JsonRawSchema, but it was: "
+                                + jsonSchema.rootElement().getClass());
+            }
+
+            Map<String, Object> schemaMap = toMap(jsonSchema.rootElement(), strict);
+            var format = new HashMap<String, Object>();
+            format.put(FIELD_TYPE, TYPE_JSON_SCHEMA);
+            if (jsonSchema.name() != null) {
+                format.put(FIELD_NAME, jsonSchema.name());
+            }
+            var jsonSchemaConfig = new HashMap<String, Object>();
+            jsonSchemaConfig.put(FIELD_SCHEMA, schemaMap);
+            jsonSchemaConfig.put(FIELD_STRICT, strict);
+            if (jsonSchema.name() != null) {
+                jsonSchemaConfig.put(FIELD_NAME, jsonSchema.name());
+            }
+            format.put(FIELD_SCHEMA, schemaMap);
+            format.put(FIELD_JSON_SCHEMA, jsonSchemaConfig);
+            textConfig.put(FIELD_FORMAT, format);
+        }
+
+        return textConfig;
+    }
+
+    @Override
+    public ChatRequestParameters defaultRequestParameters() {
+        return DefaultChatRequestParameters.builder()
+                .modelName(modelName)
+                .temperature(temperature)
+                .topP(topP)
+                .maxOutputTokens(maxOutputTokens)
+                .build();
+    }
+
+    @Override
+    public List<ChatModelListener> listeners() {
+        return Collections.unmodifiableList(listeners);
```

Comment:
```suggestion
        return listeners;
```
it is already unmodifiable